proposals/async-method-builders.md

```diff
@@ -30,17 +36,31 @@ We need a way to have an individual async method opt-in to a specific builder.
 ## Detailed design
 [design]: #detailed-design
 
-#### P0: AsyncMethodBuilderAttribute applicable to methods
+#### P0: AsyncMethodBuilderOverrideAttribute applied on async methods
 
-In dotnet/runtime, change `AsyncMethodBuilderAttribute`'s AttributeUsage from:
-```C#
-AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Delegate | AttributeTargets.Enum
-```
-to also include Method:
-```C#
-AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Delegate | AttributeTargets.Enum | AttributeTargets.Method
+In `dotnet/runtime`, add `AsyncMethodBuilderOverrideAttribute`:
+```csharp
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Indicates the type of the async method builder that should be used by a language compiler to
+    /// build the attributed method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface | AttributeTargets.Method | AttributeTargets.Module, Inherited = false, AllowMultiple = false)]
+    public sealed class AsyncMethodBuilderOverrideAttribute : Attribute
+    {
+        /// <summary>Initializes the <see cref="AsyncMethodBuilderOverrideAttribute"/>.</summary>
+        /// <param name="builderType">The <see cref="Type"/> of the associated builder.</param>
+        public AsyncMethodBuilderOverrideAttribute(Type builderType) => BuilderType = builderType;
+
+        // for scoped application (use property for targetReturnType? have compiler check that it's provided)
+        public AsyncMethodBuilderOverrideAttribute(Type builderType, Type targetReturnType) => ...
+
+        /// <summary>Gets the <see cref="Type"/> of the associated builder.</summary>
+        public Type BuilderType { get; }
```

Comment:
If we're adding that second ctor above (which is more of a P1), we'd probably also want a property here returning it.